agent/proxycfg/connect_proxy.go

```diff
@@ -326,6 +327,18 @@ func (s *handlerConnectProxy) handleUpdate(ctx context.Context, u cache.UpdateEv
 				delete(snap.ConnectProxy.WatchedDiscoveryChains, uid)
 			}
 		}
+		for uid := range snap.ConnectProxy.PassthroughUpstreams {
+			if _, ok := seenUpstreams[uid]; !ok {
+				delete(snap.ConnectProxy.PassthroughUpstreams, uid)
+
+				for addr, indexed := range snap.ConnectProxy.PassthroughIndices {
```

Comment:
Should you separately do a range over the PassthroughIndices map to do a belt-and-suspenders cleanup of stray references too?

Reply:
Yeah, breaking this off